nucliadb/nucliadb/search/search/query.py

```diff
@@ -89,10 +90,20 @@ async def global_query_to_pb(
     request.min_score = min_score
     request.body = query
     request.with_duplicates = with_duplicates
-    request.filter.tags.extend(filters)
+    if len(filters) > 0:
+        request.filter.tags.extend(filters)
+        node_features.inc({"type": "filters"})
+        if any(tag.startswith("/l/") for tag in filters):
```

Comment:
nit (micro improvement -- maybe its overkill if the request has very few tags ) : you are iterating twice on the filter list to detect the two prefixes, and on the second pass you are trying to detect the `/e/` prefix on tags that were potentially detected as being `/l/`.

A single *sorted* loop will be slightly more efficient because the prefixed tag will be at the beginning of the list, so if you iterate and find both prefixes, you can break. And you can also break once the first char is not `/` anymore.

untested:

```
>>> def detect_prefixes(tags):
...     tags.sort()
...     e_prefix = False
...     t_prefix = False
...     for tag in tags:
...         if tag[0] != '/' or (e_prefix and t_prefix):
...             break
...         if tag.startswith('/e/'):
...             e_prefix = True
...             # set the inc
...         elif tag.startwith('/t'):
...             t_prefix = True
...             # set the inc
```

Also: maybe `/l/` and `/e/` can be constants

Reply:
yup, agree! It's not that hard to make this metrics code as lightweight as possible, so let's do it!